dvc/remote/gdrive.py

```diff
@@ -195,10 +207,28 @@ def drive(self):
 
             # Pass non existent settings path to force DEFAULT_SETTINGS loading
             gauth = GoogleAuth(settings_file="")
-            gauth.CommandLineAuth()
 
-            if os.getenv(RemoteGDrive.GDRIVE_USER_CREDENTIALS_DATA):
-                os.remove(self.gdrive_user_credentials_path)
+            try:
+                gauth.CommandLineAuth()
+            except RefreshError as e:
+                raise GDriveAccessTokenRefreshError(
+                    msg="Google Drive's access token refreshment is failed",
+                    cause=e,
+                )
+            except KeyError as e:
+                raise GDriveMissedCredentialKeyError(
+                    msg="Google Drive's user credentials file '{}' "
+                    "misses value for key '{}'".format(
+                        self.gdrive_user_credentials_path, str(e)
+                    )
+                )
+            except (AuthenticationError, Exception) as e:
+                raise DvcException(
+                    "Google Drive authentication failed", cause=e
+                )
+            finally:
+                if os.getenv(RemoteGDrive.GDRIVE_USER_CREDENTIALS_DATA):
+                    os.remove(self.gdrive_user_credentials_path)
```

Comment:
Sorry for not asking earlier, but could you please explain why are we deleting gdrive_user_credentials_path on that env var? I might've missed an explanation somewhere earlier.

Reply:
if `GDRIVE_USER_CREDENTIALS_DATA` is set (it should contain gdrive user access token which gives direct access to the user's gdrive), the content is written to `gdrive_user_credentials_path` to be used by PyDrive on auth. Deleting of `gdrive_user_credentials_path` happens after authorization to not keep sensitive information anywhere on disk.

Reply:
Got it. Thanks!